requirements.txt

```diff
@@ -1,3 +1,2 @@
 torch>=2.2.0
 lightning @ git+https://github.com/Lightning-AI/lightning@ed367ca675861cdf40dbad2e4d66f7eee2ec50af
-jsonargparse[signatures]  # CLI
```

Comment:
Removing this here means you can now no longer use any of the scripts without installing all.txt. What is the point of making this optional if using lit-gpt needs this anyway for any meaningful use?

Reply:
I consider `requirements.txt` as the requirements for the package and `requirements-all.txt` as the requirements for the scripts.

The package (model definition, tokenizer, etc) doesn't need jsonargparse because it's not used. It's only used when called as a script

Most CLI users already had to install all requirements for `sentencepiece` and `tokenizers`